stages/org.osbuild.gcp.guest-agent.conf

```diff
@@ -0,0 +1,269 @@
+    "minProperties": 1,
+    "properties": {
+      "ethernet_proto_id": {
+        "type": "string",
+        "description": "Protocol ID string for daemon added routes."
+      },
+      "ip_aliases": {
+        "type": "boolean",
+        "description": "Disables setting up alias IP routes."
+      },
+      "target_instance_ips": {
+        "type": "boolean",
+        "description": "Disables internal IP address load balancing."
+      }
+    }
+  },
+  "MetadataScripts": {
+    "type": "object",
+    "additionalProperties": false,
+    "description": "MetadataScripts section.",
+    "minProperties": 1,
+    "properties": {
+      "default_shell": {
+        "type": "string",
+        "description": "String with the default shell to execute scripts."
+      },
+      "run_dir": {
+        "type": "string",
+        "description": "String base directory where metadata scripts are executed."
+      },
+      "startup": {
+        "type": "boolean",
+        "description": "Disables startup script execution."
+      },
+      "shutdown": {
+        "type": "boolean",
+        "description": "Disables shutdown script execution."
+      }
+    }
+  },
+  "NetworkInterfaces": {
+    "type": "object",
+    "additionalProperties": false,
+    "description": "NetworkInterfaces section.",
+    "minProperties": 1,
+    "properties": {
+      "setup": {
+        "type": "boolean",
+        "description": "Skips network interface setup."
+      },
+      "ip_forwarding": {
+        "type": "boolean",
+        "description": "Skips IP forwarding."
+      },
+      "dhcp_command": {
+        "type": "string",
+        "description": "Path for alternate dhcp executable used to enable network interfaces."
+      }
+    }
+  }
+},
+"additionalProperties": false,
+"required": ["config"],
+"description": "Configure GCP guest-agent.",
+"properties": {
+  "config_scope": {
+    "type": "string",
+    "description": "Create distro-wide or instance-specific configuration.",
+    "enum": ["distro", "instance"],
+    "default": "distro"
+  },
+  "config": {
+    "type": "object",
+    "additionalProperties": false,
+    "description": "GCP guest-agent configuration.",
+    "minProperties": 1,
+    "properties": {
+      "Accounts": {
+        "$ref": "#/definitions/Accounts"
+      },
+      "Daemons": {
+        "$ref": "#/definitions/Daemons"
+      },
+      "InstanceSetup": {
+        "$ref": "#/definitions/InstanceSetup"
+      },
+      "IpForwarding": {
+        "$ref": "#/definitions/IpForwarding"
+      },
+      "MetadataScripts": {
+        "$ref": "#/definitions/MetadataScripts"
+      },
+      "NetworkInterfaces": {
+        "$ref": "#/definitions/NetworkInterfaces"
+      }
+    }
+  }
+}
+"""
+
+def option_value_to_str(value):
+    """
+    Convert allowed types of option values to string.
+
+    The instance_configs.cfg allows comma seperated string lists as an option value.
+    """
+    if isinstance(value, list):
+        value = ", ".join(value)
```

Comment:
I was not sure if adding the extra space here is OK, because I would expect that values will be joined without the space if they are supposed to be comma-separated. However the `go-ini` package used by the GCP guest-agent to read the configuration file is trimming space when parsing values separated by a given delimiter, so this should be OK:
https://github.com/go-ini/ini/blob/6098d4257e883d8cf846e9012bdd5b03811af4ce/key.go#L528